src/dao/pages/UpdateDAODetails/UpdateDAODetails.tsx

```diff
@@ -6,18 +6,20 @@ import { Wizard, Color, LoadingDialog } from "@shared/ui-kit";
 import { WarningIcon } from "@chakra-ui/icons";
 import { SettingsForm } from "./types";
 import { useHandleTransactionSuccess } from "@dex/hooks";
-import { useDAOs, useUpdateDAODetails } from "@dao/hooks";
+import { useContract, useDAOs, useUpdateDAODetails } from "@dao/hooks";
 import { getDAOLinksRecordArray } from "../utils";
 import { isEmpty } from "ramda";
 
 export function UpdateDAODetails() {
   const { accountId: daoAccountId = "" } = useParams();
+  const daoAccountIdQueryResults = useContract(daoAccountId);
+  const daoAccountEVMAddress = daoAccountIdQueryResults.data?.data.evm_address;
   const location = useLocation();
   const currentDaoType = location.pathname.split("/").at(2) ?? "";
   const backTo = `/${currentDaoType}/${daoAccountId}/settings`;
-  const daosQueryResults = useDAOs(daoAccountId);
+  const daosQueryResults = useDAOs();
   const { data: daos } = daosQueryResults;
-  const dao = daos?.find((dao) => dao.accountId === daoAccountId);
+  const dao = daos?.find((dao) => dao.accountId.toLowerCase() == daoAccountEVMAddress?.toLowerCase());
```

Comment:
Use [Strict equality (===)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality) not [Equality (==)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality). The Mozilla developer doc links explain the difference. There is rarely ever a reason to use `==` for typical value equality checks.

Reply:
Done.